File: front/src/services/cart/CartService.js
import axios from "axios";

const baseURL = "http://localhost:8000/api";

// 장바구니에 추가
const addToCart = (data) => {
  return axios.post(baseURL + `/cart/add`, data);
};

// 이메일마다 장바구니 전체조회
const getItemsFromEmail = (userEmail) => {
  return axios.get(baseURL + `/cart/detail/${userEmail}`);
};

// 장바구니 개별 삭제
const deleteFromCartId = (cartId) => {
  return axios.delete(baseURL + `/cart/delete/${cartId}`);
};

// 장바구니 전체 삭제
const deleteFromCartEmail = (userEmail) => {
  return axios.delete(baseURL + `/cart/delete/all/${userEmail}`);
};

// 장바구니 개수 카운트
const countCartItems = (userEmail) => {
  return axios.get(baseURL + `/cart/count/${userEmail}`);
};

// cartId로 장바구니 상세조회
const getItemsFromCartId = (cartId) => {
  return axios.get(baseURL + `/cart/detail/item/${cartId}`);
};

const CartService = {
  addToCart,
  getItemsFromEmail,
  deleteFromCartId,
  deleteFromCartEmail,
  countCartItems,
  getItemsFromCartId,
};

export default CartService;

File: SimpleDms/src/main/java/com/simplecoding/simpledms/service/cart/CartService.java
package com.simplecoding.simpledms.service.cart;

import com.simplecoding.simpledms.mapper.cart.CartMapper;
import com.simplecoding.simpledms.vo.cart.Cart;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class CartService {

    private final CartMapper cartMapper;

    // 장바구니에 추가
    public void addToCart(Cart cart) {
        cartMapper.addToCart(cart);
    }

    // 이메일로 상세조회해서 장바구니 출력: 배열
    public List<Cart> getItemsFromEmail(String userEmail) {
        return cartMapper.getItemsFromEmail(userEmail);
    }

    // 개별 삭제
    public void deleteFromCartId(int cartId) {
        cartMapper.deleteFromCartId(cartId);
    }

    // 전체 삭제
    public void deleteFromEmail(String userEmail) {
        cartMapper.deleteFromEmail(userEmail);
    }

    // 장바구니에 담긴 개수 세기
    public int countCartItems(String userEmail) {
        return cartMapper.countCartItems(userEmail);
    }


}
